pkg/clients/connectioncache/connectioncache_impl.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package connectioncache
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/grpc"
+
+	"storj.io/storj/pkg/clients/transportclient"
+	"storj.io/storj/pkg/overlay"
+	proto "storj.io/storj/protos/overlay"
+)
+
+//NewConnCache returns ConnCache interface
+func NewConnectionCache(ctx context.Context, addr string, tc Client) (ConnectionCache, error) {
+	switch tc {
+	case Overlay:
```

Comment:
you don't need to do this. all the connection cache does is return *grpc.ClientConns. why do you need an overlay client at all?